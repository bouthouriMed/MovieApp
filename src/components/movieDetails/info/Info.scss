@import "@/styles/variables";

.info {
  flex: 2;
  display: flex;
  flex-direction: column;
  gap: 1rem;

  h1 {
    font-size: 2.5rem;
    font-weight: bold;
  }
  .tagline {
    font-style: italic;
  }
  .meta {
    font-size: 0.95rem;
    color: lighten($text-color, 30%);
  }
  .rating {
    font-size: 1rem;
    font-weight: 600;
    color: $primary-color;
  }
  .overview {
    font-size: 1rem;
    line-height: 1.6;
    color: lighten($text-color, 20%);
  }

  @media (max-width: 768px) {
    h1 {
      font-size: 2rem;
    }
    .overview {
      font-size: 0.95rem;
      line-height: 1.5;
    }
    .meta,
    .rating {
      font-size: 0.9rem;
    }
  }

  @media (max-width: 480px) {
    h1 {
      font-size: 1.75rem;
    }
    .overview {
      font-size: 0.9rem;
    }
    .meta,
    .rating {
      font-size: 0.85rem;
    }
  }
}
