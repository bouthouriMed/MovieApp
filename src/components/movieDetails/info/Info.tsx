import { memo, ReactNode } from "react";
import { Movie } from "src/pages/movieDetailPage/types";
import "./Info.scss";

interface InfoProps {
  movie: Movie;
  children?: ReactNode;
}

const Info = ({ movie, children }: InfoProps) => {
  return (
    <div className="info">
      <h1>{movie.title}</h1>
      {movie.tagline && <h3 className="tagline">“{movie.tagline}”</h3>}

      <div className="meta">
        <span>{movie.release_date.slice(0, 4)}</span> •{" "}
        <span>{movie.runtime} min</span> •{" "}
        <span>{movie.genres.map((g) => g.name).join(", ")}</span>
      </div>

      <div className="rating">
        ⭐ {movie.vote_average.toFixed(1)} / 10 ({movie.vote_count} votes)
      </div>

      <p className="overview">{movie.overview}</p>

      {children}
    </div>
  );
};

export default memo(Info);
