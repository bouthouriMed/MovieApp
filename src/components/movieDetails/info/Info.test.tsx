import { render, screen } from "@testing-library/react";
import { mockedMovie } from "@/__mocks__/test_mocks";
import Info from "./Info";

describe("Info Component", () => {
  it("renders movie title, tagline, meta, rating, and overview", () => {
    render(<Info movie={mockedMovie} />);

    expect(screen.getByText(mockedMovie.title)).toBeInTheDocument();
    expect(screen.getByText(/Test tagline/)).toBeInTheDocument();
    expect(screen.getByText("2023")).toBeInTheDocument();
    expect(screen.getByText("120 min")).toBeInTheDocument();
    expect(screen.getByText("Action")).toBeInTheDocument();
    expect(screen.getByText(/8.0 \/ 10 \(100 votes\)/)).toBeInTheDocument();
    expect(screen.getByText(/Test overview/)).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = render(<Info movie={mockedMovie} />);
    expect(container).toMatchSnapshot();
  });
});
