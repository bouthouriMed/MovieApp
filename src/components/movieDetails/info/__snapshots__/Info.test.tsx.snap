// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Info Component > matches snapshot 1`] = `
<div>
  <div
    class="info"
  >
    <h1>
      Test Movie
    </h1>
    <h3
      class="tagline"
    >
      “
      Test tagline
      ”
    </h3>
    <div
      class="meta"
    >
      <span>
        2023
      </span>
       •
       
      <span>
        120
         min
      </span>
       •
       
      <span>
        Action
      </span>
    </div>
    <div
      class="rating"
    >
      ⭐ 
      8.0
       / 10 (
      100
       votes)
    </div>
    <p
      class="overview"
    >
      Test overview
    </p>
  </div>
</div>
`;
